components/studio/test/shared/studio-enter.exp

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env expect
+
+set env(HAB_NOCOLORING) true
+set studio_enter_command $::env(studio_enter_command)
+# Print out some helpful tracing messages in the test output.
+proc log {message} {
+    puts "LOG >>> $message"
+}
+
+# Cleanup after the test
+exit -onexit {
+    exec hab studio rm
+}
+
+# Installing packages can take time
+set timeout 30
+
+# I HAVE NO IDEA WHAT THAT SYNTAX MEANS BUT IT IS REQUIRED :/
+spawn {*}$studio_enter_command
+expect_tty {
+  {[1][default:/src:0]} { 
+    log "Studio entered successfully"
+  }
+  {default} { log "Something odd" }
+  eof {
+    error "Unable to enter studio"
+  }
+  timeout {
+    error "Timeout entering studio"
+  } 
+}
+
+send "test/shared/studio-internals/test-all.sh\n"
```

Comment:
Not quite sure what's going on yet, but it appears that this is being run both inside the studio, as well as outside the studio when the first run is done 😱

Reply:
Interesting, I wasn't observing that behavior.   How were you invoking the tests?  It's also entirely possible I messed something up when I extracted this from my studio work branch 😂  I'll look into it.